File: BackEnd/Categories/CreateCategory/src/config/env.js
require('dotenv').config(); // Cargar las variables de entorno desde .env

const config = {
  mongoURI: process.env.MONGO_URI || 'mongodb://localhost:27017/category_db',
  appPort: process.env.APP_PORT || 8086,
};

module.exports = config;

File: BackEnd/Domains/Products/GetProductsByCategory/config/env.go
package config

import (
	"log"
	"os"

	"github.com/joho/godotenv"
)

// LoadEnv carga las variables de entorno desde un archivo .env
func LoadEnv() {
	err := godotenv.Load()
	if err != nil {
		log.Println("No .env file found. Using system environment variables.")
	}
}

// GetEnv obtiene el valor de una variable de entorno con un valor por defecto
func GetEnv(key, defaultValue string) string {
	value := os.Getenv(key)
	if value == "" {
		return defaultValue
	}
	return value
}

// GetPort retrieves the application port from environment variables.
func GetPort() string {
	port := os.Getenv("APP_PORT")
	if port == "" {
		port = "80" // Default port
	}
	return port
}
